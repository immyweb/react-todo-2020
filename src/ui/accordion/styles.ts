import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
