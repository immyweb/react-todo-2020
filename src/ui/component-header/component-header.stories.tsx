// import * as React from 'react';

// import { storiesOf } from '@storybook/react';

// import { SunColSix } from '../../styles/grids';
// import { ComponentHeader } from '../component-header/';

// import { colours } from '../../styles/colours';

// const stories = storiesOf('Component Header', module);

// const theme = {
//   primary: colours.$fabulous,
//   secondary: colours.$fabulousSecondary,
// };

// stories.add('default useage', () => (
//   <SunColSix>
//     <ComponentHeader
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//     />
//   </SunColSix>
// ));

// stories.add('inline useage', () => (
//   <SunColSix>
//     <ComponentHeader
//       inline
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//     />
//   </SunColSix>
// ));

// stories.add('center align', () => (
//   <SunColSix>
//     <ComponentHeader
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//       alignment={{ x: 'center' }}
//     />
//   </SunColSix>
// ));

// stories.add('right align', () => (
//   <SunColSix>
//     <ComponentHeader
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//       alignment={{ x: 'right' }}
//     />
//   </SunColSix>
// ));

// stories.add('inline center', () => (
//   <SunColSix>
//     <ComponentHeader
//       inline
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//       alignment={{ x: 'center' }}
//     />
//   </SunColSix>
// ));

// stories.add('inline right', () => (
//   <SunColSix>
//     <ComponentHeader
//       inline
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//       alignment={{ x: 'right' }}
//     />
//   </SunColSix>
// ));

// stories.add('title only', () => (
//   <SunColSix>
//     <ComponentHeader title="rest of headline" />
//   </SunColSix>
// ));

// stories.add('highlight only', () => (
//   <SunColSix>
//     <ComponentHeader highlight="highlight" />
//   </SunColSix>
// ));

// stories.add('highlight only inline', () => (
//   <SunColSix>
//     <ComponentHeader highlight="highlight" inline />
//   </SunColSix>
// ));

// stories.add('subTitle only', () => (
//   <SunColSix>
//     <ComponentHeader subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus" />
//   </SunColSix>
// ));

// stories.add('with link', () => (
//   <SunColSix>
//     <ComponentHeader
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, <a href='http://www.google.com' target='_blank'>consectetur adipiscing</a> elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//     />
//   </SunColSix>
// ));

// stories.add('with theme', () => (
//   <SunColSix>
//     <ComponentHeader
//       highlight="highlight"
//       title="rest of headline"
//       subTitle="Lorem ipsum dolor sit amet, <a href='http://www.google.com' target='_blank'>consectetur adipiscing</a> elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
//       theme={theme}
//     />
//   </SunColSix>
// ));

// stories.add('with background', () => (
//   <SunColSix>
//     <ComponentHeader title="rest of headline" theme={theme} background />
//   </SunColSix>
// ));
