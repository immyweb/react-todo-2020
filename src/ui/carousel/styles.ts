import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
