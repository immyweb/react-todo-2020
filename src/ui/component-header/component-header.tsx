import * as React from 'react';

import { TitleWrapper, Highlight, Title, SubTitle } from './styles';
import { fonts } from '../../styles/fonts';

import { colours } from '../../styles/colours';
import { ITheme } from '../../types/theme';

import { IAlign } from '../../utils/alignment-utils';
import { dangerouslySet } from '../../utils/html-utils';

interface IProps {
  className?: string;
  inline?: boolean;
  highlight?: string;
  title?: string;
  subTitle?: string;
  theme?: ITheme;
  alignment?: IAlign;
  background?: boolean;
}

const defaultProps = {
  inline: false,
  theme: {
    primary: colours.$main,
    secondary: colours.$mainSecondary,
  },
  alignment: {
    x: 'left',
  },
  background: false,
};

export const ComponentHeader: React.SFC<IProps> = ({
  className,
  inline,
  highlight,
  title,
  subTitle,
  theme,
  alignment,
  background,
}) => {
  const _inline = inline ? inline : defaultProps.inline;
  const _theme = theme ? theme : defaultProps.theme;
  const _alignment = alignment ? alignment : defaultProps.alignment;
  const _background = background ? background : defaultProps.background;

  const returnText = () => {
    if (background) {
      return (
        <Title
          fontName={fonts.$theSunHeavyCondensed}
          inline={_inline}
          alignment={_alignment}
          background={_background}
        >
          {title}
        </Title>
      );
    }

    return (
      <div>
        {highlight && (
          <Highlight
            fontName={fonts.$theSunHeavyCondensed}
            fontColour={_theme.primary}
            inline={_inline}
            alignment={_alignment}
            background={_background}
          >
            {highlight}
          </Highlight>
        )}
        <Title
          fontName={fonts.$theSunHeavyCondensed}
          inline={_inline}
          alignment={_alignment}
          background={_background}
        >
          {title}
        </Title>
      </div>
    );
  };

  return (
    <div className={className}>
      <TitleWrapper
        theme={theme}
        inline={_inline}
        alignment={_alignment}
        background={_background}
        className={`${className}-inner`}
      >
        {returnText()}
      </TitleWrapper>
      {subTitle && !background && (
        <SubTitle
          className={`${className}-subtitle`}
          linkColour={_theme.primary}
          alignment={_alignment}
          dangerouslySetInnerHTML={dangerouslySet(`${subTitle}`)}
          hasBackground={_background}
        />
      )}
    </div>
  );
};
