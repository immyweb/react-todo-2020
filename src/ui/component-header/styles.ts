import glamorous from 'glamorous';

import { P, H2 } from '../typography/typography';
import { colours } from '../../styles/colours';

import { mediaQueries } from '../../styles/breakpoints';
import { IAlign } from '../../utils/alignment-utils';

interface IGProps {
  inline?: boolean;
  alignment?: IAlign;
  linkColour?: string;
  background?: boolean;
  hasBackground?: boolean;
}

const styles = {
  fontSizeLarge: 28,
  fontSizeSmall: 24,
  inlineHighlight: 16,
  marginBottomLarge: 12,
  h2LineHeight: 1.14,
  pLineHeight: 1.38,
};

export const TitleWrapper = glamorous.div<IGProps>(
  {
    label: 'title-wrapper',
  },
  ({ alignment, background, theme }: any) => ({
    textAlign: alignment.x,
    background: background ? theme.primary : 'none',
    padding: background ? '13px 20px 13px 15px' : 0,
    borderLeftStyle: background ? 'solid' : 'none',
    borderLeftWidth: '5px',
    borderLeftColor: theme.secondary,
  }),
);

export const Title = glamorous(H2)<IGProps>(
  {
    label: 'title',
    fontSize: styles.fontSizeSmall,
    lineHeight: styles.h2LineHeight,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      lineHeight: styles.h2LineHeight,
    },
  },
  ({ inline, alignment, background }: any) => ({
    display: inline ? 'inline-block' : 'block',
    textTransform: background ? 'unset' : inline ? 'unset' : 'uppercase',
    marginBottom: background ? 0 : styles.marginBottomLarge,
    textAlign: alignment.x,
    color: background ? colours.$white : colours.$offBlack,
  }),
);

export const Highlight = glamorous(Title)<IGProps>(
  {
    label: 'highlight',
    marginBottom: 0,
    fontSize: styles.fontSizeSmall,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      marginBottom: 0,
    },
  },
  ({ inline, alignment, fontColour, background }: any) => ({
    fontSize: inline ? styles.fontSizeSmall : styles.inlineHighlight,
    textTransform: 'uppercase',
    textAlign: alignment.x,
    color: fontColour,
    marginRight: inline ? 4 : 0,
    [mediaQueries.tabletPortrait]: {
      fontSize: inline ? styles.fontSizeLarge : styles.inlineHighlight,
    },
  }),
);

export const SubTitle = glamorous(P)<IGProps>(
  {
    label: 'sub-title',
    lineHeight: styles.pLineHeight,
    textTransform: 'lowercase',
    marginBottom: 20,
    [mediaQueries.tabletPortrait]: {
      lineHeight: styles.pLineHeight,
    },
  },
  ({ alignment, linkColour, hasBackground }: any) => ({
    textAlign: alignment.x,
    marginTop: hasBackground ? 20 : 0,
    '& a': {
      color: linkColour,
      textDecoration: 'none',
    },

    '& a:hover': {
      textDecoration: 'underline',
    },
  }),
);
