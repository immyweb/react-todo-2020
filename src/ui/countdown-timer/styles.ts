import glamorous from 'glamorous';
import { H1, H2, P } from '../typography/typography';
import { mediaQueries } from '../../styles/breakpoints';
import { fonts } from '../../styles/fonts';

import { IAlign } from '../../utils/alignment-utils';

export const CountdownWrapper = glamorous.div<{
  backgroundImage?: string;
}>(
  {
    position: 'relative',
    width: '100%',
    height: '100%',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    backgroundPosition: 'center center',
    minHeight: 185,
  },
  ({ backgroundImage }) => ({
    backgroundImage: backgroundImage ? `url(${backgroundImage})` : '',
  }),
);

export const CountdownContent = glamorous.div<{ alignment?: IAlign }>(
  {
    position: 'absolute',
    width: '100%',
    padding: 10,
  },
  ({ alignment }) => ({
    top: alignment ? alignment.y : 0,
    left: alignment ? alignment.x : 0,
    transform: `translate(-${alignment ? alignment.x : 0}, -${alignment ? alignment.y : 0})`,
  }),
);

export const CountdownFrame = glamorous.div({
  minHeight: 80,
  position: 'relative',
});

export const CountdownAligner = glamorous.div<{ alignment?: IAlign }>(
  {
    position: 'absolute',
    width: '100%',
    top: '50%',
  },
  ({ alignment }) => ({
    left: alignment ? alignment.x : 0,
    transform: `translate(-${alignment ? alignment.x : 0}, -50%)`,
  }),
);

export const HeaderWrapper = glamorous.div({
  label: 'header-wrapper',
});

export const Title = glamorous(H2)<{ alignment?: IAlign; textColour: string }>(
  {
    marginBottom: 2,
    textTransform: 'uppercase',
    [mediaQueries.tabletPortrait]: {
      lineHeight: 1,
    },
  },
  ({ alignment, textColour }: any) => ({
    textAlign: alignment ? alignment.x : 'left',
    color: textColour,
  }),
);

export const SubTitle = glamorous(P)<{
  alignment?: IAlign;
  textColour: string;
}>(
  {
    fontSize: 14,
    lineHeight: 1.25,
    marginBottom: 5,
    [mediaQueries.tabletPortrait]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
  },
  ({ alignment, textColour }: any) => ({
    textAlign: alignment ? alignment.x : 'left',
    color: textColour,
  }),
);

export const UnitWrapper = glamorous.div<{
  alignment?: IAlign;
  opacity?: number;
}>(
  {
    transition: `opacity ${500}ms linear`,
    transitionDelay: `${1000}ms`,
    ':nth-child(n) > div': {
      marginRight: 3,
    },
  },
  ({ alignment, opacity }: any) => ({
    opacity: opacity ? opacity : 0,
    textAlign: alignment ? alignment.x : 'left',
  }),
);

export const UnitInstance = glamorous.div({
  display: 'inline-block',
});

export const UnitValue = glamorous(H1)<{ textColour: string }>(
  {
    fontSize: 42,
    lineHeight: 1.17,
    fontFamily: fonts.$theSunHeavyNarrow,
    textAlign: 'center',
    marginBottom: 0,
    minWidth: 47,
    [mediaQueries.tabletPortrait]: {
      fontSize: 48,
      lineHeight: 1.17,
    },
    [mediaQueries.mobile]: {
      minWidth: 62,
      fontSize: 48,
      lineHeight: 1.17,
    },
  },
  ({ textColour }) => ({
    color: textColour,
  }),
);

export const UnitLabel = glamorous(P)<{ textColour: string }>(
  {
    fontSize: 12,
    lineHeight: 1.25,
    marginBottom: 0,
    textAlign: 'center',
    [mediaQueries.tabletPortrait]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
    [mediaQueries.mobile]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
  },
  ({ textColour }) => ({
    color: textColour,
  }),
);

export const DigitContainer = glamorous.div({
  width: 29,
  display: 'inline-block',
  textAlign: 'center',
});
