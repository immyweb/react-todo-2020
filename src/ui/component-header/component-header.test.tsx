import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { ComponentHeader } from './component-header';

import { TitleWrapper, Highlight, Title, SubTitle } from './styles';

import { colours } from '../../styles/colours';

const theme = {
  primary: colours.$fabulous,
  secondary: colours.$fabulousSecondary,
};

const shallowComponentHeader = () => {
  return shallow(
    <ComponentHeader
      inline
      highlight="highlight"
      title="rest of headline"
      subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
      alignment={{ x: 'left' }}
      theme={theme}
    />,
  );
};

const mountComponentHeaderBackground = () => {
  return mount(
    <ComponentHeader
      inline
      highlight="highlight"
      title="rest of headline"
      subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
      alignment={{ x: 'left' }}
      theme={theme}
      background={true}
    />,
  );
};

const mountComponentHeader = () => {
  return mount(
    <ComponentHeader
      inline
      highlight="highlight"
      title="rest of headline"
      subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
      alignment={{ x: 'left' }}
      theme={theme}
    />,
  );
};

describe('<ComponentHeader />', () => {
  it('renders correctly', () => {
    const wrapper = shallowComponentHeader();
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('has a inline prop', () => {
    const wrapper = mountComponentHeader();
    expect(wrapper.prop('inline')).toEqual(true);
  });
  it('has a theme prop', () => {
    const wrapper = mountComponentHeader();
    expect(wrapper.prop('theme')).toEqual({
      primary: colours.$fabulous,
      secondary: colours.$fabulousSecondary,
    });
  });
  it('has alignment prop', () => {
    const wrapper = mountComponentHeader();
    expect(wrapper.prop('alignment')).toEqual({ x: 'left' });
  });

  it('has a background prop', () => {
    const wrapper = mountComponentHeaderBackground();
    expect(wrapper.prop('background')).toEqual(true);
  });

  it('has a Highlight', () => {
    const wrapper = shallowComponentHeader();
    const highlight = wrapper.find(Highlight);
    expect(highlight).toHaveLength(1);
  });
  it('has a Title', () => {
    const wrapper = shallowComponentHeader();
    const title = wrapper.find(Title);
    expect(title).toHaveLength(1);
  });
  it('has a TitleWrapper', () => {
    const wrapper = shallowComponentHeader();
    const titleWrapper = wrapper.find(TitleWrapper);
    expect(titleWrapper).toHaveLength(1);
  });
  it('has a SubTitle', () => {
    const wrapper = shallowComponentHeader();
    const subTitle = wrapper.find(SubTitle);
    expect(subTitle).toHaveLength(1);
  });
});
