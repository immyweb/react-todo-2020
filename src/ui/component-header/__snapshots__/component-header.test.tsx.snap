// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentHeader /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    alignment={
      Object {
        "x": "left",
      }
    }
    highlight="highlight"
    inline={true}
    subTitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus"
    theme={
      Object {
        "primary": "#e665bf",
        "secondary": "#a6498a",
      }
    }
    title="rest of headline"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <glamorous(div)
          alignment={
            Object {
              "x": "left",
            }
          }
          background={false}
          className="undefined-inner"
          inline={true}
          theme={
            Object {
              "primary": "#e665bf",
              "secondary": "#a6498a",
            }
          }
        >
          <div>
            <glamorous(glamorous(glamorous(h2)))
              alignment={
                Object {
                  "x": "left",
                }
              }
              background={false}
              fontColour="#e665bf"
              fontName="TheSun-HeavyCondensed"
              inline={true}
            >
              highlight
            </glamorous(glamorous(glamorous(h2)))>
            <glamorous(glamorous(h2))
              alignment={
                Object {
                  "x": "left",
                }
              }
              background={false}
              fontName="TheSun-HeavyCondensed"
              inline={true}
            >
              rest of headline
            </glamorous(glamorous(h2))>
          </div>
        </glamorous(div)>,
        <glamorous(glamorous(glamorous.P))
          alignment={
            Object {
              "x": "left",
            }
          }
          className="undefined-subtitle"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus",
            }
          }
          hasBackground={false}
          linkColour="#e665bf"
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignment": Object {
            "x": "left",
          },
          "background": false,
          "children": <div>
            <glamorous(glamorous(glamorous(h2)))
              alignment={
                Object {
                  "x": "left",
                }
              }
              background={false}
              fontColour="#e665bf"
              fontName="TheSun-HeavyCondensed"
              inline={true}
            >
              highlight
            </glamorous(glamorous(glamorous(h2)))>
            <glamorous(glamorous(h2))
              alignment={
                Object {
                  "x": "left",
                }
              }
              background={false}
              fontName="TheSun-HeavyCondensed"
              inline={true}
            >
              rest of headline
            </glamorous(glamorous(h2))>
          </div>,
          "className": "undefined-inner",
          "inline": true,
          "theme": Object {
            "primary": "#e665bf",
            "secondary": "#a6498a",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <glamorous(glamorous(glamorous(h2)))
                alignment={
                  Object {
                    "x": "left",
                  }
                }
                background={false}
                fontColour="#e665bf"
                fontName="TheSun-HeavyCondensed"
                inline={true}
              >
                highlight
              </glamorous(glamorous(glamorous(h2)))>,
              <glamorous(glamorous(h2))
                alignment={
                  Object {
                    "x": "left",
                  }
                }
                background={false}
                fontName="TheSun-HeavyCondensed"
                inline={true}
              >
                rest of headline
              </glamorous(glamorous(h2))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alignment": Object {
                  "x": "left",
                },
                "background": false,
                "children": "highlight",
                "fontColour": "#e665bf",
                "fontName": "TheSun-HeavyCondensed",
                "inline": true,
              },
              "ref": null,
              "rendered": "highlight",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alignment": Object {
                  "x": "left",
                },
                "background": false,
                "children": "rest of headline",
                "fontName": "TheSun-HeavyCondensed",
                "inline": true,
              },
              "ref": null,
              "rendered": "rest of headline",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignment": Object {
            "x": "left",
          },
          "className": "undefined-subtitle",
          "dangerouslySetInnerHTML": Object {
            "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus",
          },
          "hasBackground": false,
          "linkColour": "#e665bf",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <glamorous(div)
            alignment={
              Object {
                "x": "left",
              }
            }
            background={false}
            className="undefined-inner"
            inline={true}
            theme={
              Object {
                "primary": "#e665bf",
                "secondary": "#a6498a",
              }
            }
          >
            <div>
              <glamorous(glamorous(glamorous(h2)))
                alignment={
                  Object {
                    "x": "left",
                  }
                }
                background={false}
                fontColour="#e665bf"
                fontName="TheSun-HeavyCondensed"
                inline={true}
              >
                highlight
              </glamorous(glamorous(glamorous(h2)))>
              <glamorous(glamorous(h2))
                alignment={
                  Object {
                    "x": "left",
                  }
                }
                background={false}
                fontName="TheSun-HeavyCondensed"
                inline={true}
              >
                rest of headline
              </glamorous(glamorous(h2))>
            </div>
          </glamorous(div)>,
          <glamorous(glamorous(glamorous.P))
            alignment={
              Object {
                "x": "left",
              }
            }
            className="undefined-subtitle"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus",
              }
            }
            hasBackground={false}
            linkColour="#e665bf"
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignment": Object {
              "x": "left",
            },
            "background": false,
            "children": <div>
              <glamorous(glamorous(glamorous(h2)))
                alignment={
                  Object {
                    "x": "left",
                  }
                }
                background={false}
                fontColour="#e665bf"
                fontName="TheSun-HeavyCondensed"
                inline={true}
              >
                highlight
              </glamorous(glamorous(glamorous(h2)))>
              <glamorous(glamorous(h2))
                alignment={
                  Object {
                    "x": "left",
                  }
                }
                background={false}
                fontName="TheSun-HeavyCondensed"
                inline={true}
              >
                rest of headline
              </glamorous(glamorous(h2))>
            </div>,
            "className": "undefined-inner",
            "inline": true,
            "theme": Object {
              "primary": "#e665bf",
              "secondary": "#a6498a",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <glamorous(glamorous(glamorous(h2)))
                  alignment={
                    Object {
                      "x": "left",
                    }
                  }
                  background={false}
                  fontColour="#e665bf"
                  fontName="TheSun-HeavyCondensed"
                  inline={true}
                >
                  highlight
                </glamorous(glamorous(glamorous(h2)))>,
                <glamorous(glamorous(h2))
                  alignment={
                    Object {
                      "x": "left",
                    }
                  }
                  background={false}
                  fontName="TheSun-HeavyCondensed"
                  inline={true}
                >
                  rest of headline
                </glamorous(glamorous(h2))>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alignment": Object {
                    "x": "left",
                  },
                  "background": false,
                  "children": "highlight",
                  "fontColour": "#e665bf",
                  "fontName": "TheSun-HeavyCondensed",
                  "inline": true,
                },
                "ref": null,
                "rendered": "highlight",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alignment": Object {
                    "x": "left",
                  },
                  "background": false,
                  "children": "rest of headline",
                  "fontName": "TheSun-HeavyCondensed",
                  "inline": true,
                },
                "ref": null,
                "rendered": "rest of headline",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignment": Object {
              "x": "left",
            },
            "className": "undefined-subtitle",
            "dangerouslySetInnerHTML": Object {
              "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut accumsan venenatis neque quis malesuada. Sed ac porta enim, vel bibendum tellus",
            },
            "hasBackground": false,
            "linkColour": "#e665bf",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
